### PR TITLE
Add contribution guideline and code of conduct

### PR DESCRIPTION
**Pull request by re-xyr**
All current contributors are encouraged to review the contribution guideline and  propose necessary addition.

Will not be merged before appoval of @ice1000, @imkiva and @Glavo.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+You are encouraged to:
+
+- Report any bugs encountered in using Aya.
+  - Please include a minimal reproduction and the commit hash you are using to run the reproduction.
+  - Please include the error report.
+- Ask any questions about Aya in the discussion area.
+
+These are currently not accepted:
+
+- PRs, because Aya is still in rapid development.
```
[the remark this answers is not in the document]
Comment (ice1000) [reply .github/CONTRIBUTING.md:18]: Maybe also reference the Lean doc.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+You are encouraged to:
+
+- Open issues about the feature planning, implementation and other aspects of the Aya prover.
+- Open bug fixing PRs.
+- Open feature adding PRs when the addition of the feature is already agreed on by the community.
+
+You are not allowed to:
+
+- Push to the main branch, except for a few exceptions.Merging your PR into the main branch is wholly handled by the @bors bot.
```
Comment (ice1000) [inline .github/CONTRIBUTING.md:37]: Typesetting: the period needs a space after it

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+- Force-push to the main branch. This is strictly prohibited. No exception.
+- Perform other destructive and/or irreversible actions on the organization or repository settings without the maintainers' consent.
+
+Your responsibility:
```
Comment (ice1000) [inline .github/CONTRIBUTING.md:41]: One more thing: base yourself on the latest main branch if possible and make sure you look at the CI test results. Make sure you understand the purpose of using CI and address the problems it reports.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  - Commit messages should follow the format `KEYWORD: detailed message ....`.
+  - Please do not include overly fragmented commits; amending you commit and force-pushing to your PR branch is accepted and encouraged.
+- Please be collaborative and open to others' comments and review.
+  - When submitting your PR and issues, make sure to include proper tags and milestones (if applicable).
+  - Please request review from relevant members and at least one maintainer for your PR.
+- Other unethical behaviors not adhering to [the Code of Conduct](CODE_OF_CONDUCT.md) is not allowed and will be taken proper action upon by maintainers.
+
+## Maintainers
+
+Maintaners are the collaborators that are either paid by PLCT or declared to be a maintainer by other maintainers' consensus.
```
Comment (ice1000) [inline .github/CONTRIBUTING.md:54]: paid → sponsored

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+Maintainers should:
+
+- Continuously contribute to Aya's prosperity.
+- Be open-minded and willing to accept and/or discuss different opinions.
+- Always work towards consensus.
+
+Maintainers' responsibility:
+
+- For any important decisions that may affect Aya's prospect and general development, including major change in language goal, features and repository configuration, *no irreversible actions should be done before **all** maintainers reach consensus on a detailed plan and collective consent of potential consequences*. (ref. #60)
```
Comment (ice1000) [inline .github/CONTRIBUTING.md:66]: Are you sure this line will be rendered correctly? I suggest replacing the italic delimiter `*` with `_`

**Comment by ice1000** [comment]
I just realized we could configure bors with more options such as `required_approvals` or something. See https://bors.tech/documentation/ talking about bors.toml

**Comment by ice1000** [comment]
`commit_title` would be very useful

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+You are not allowed to:
+
+- Push to the main branch, except for a few exceptions.Merging your PR into the main branch is wholly handled by the @bors bot.
+- Force-push to the main branch. This is strictly prohibited. No exception.
+- Perform other destructive and/or irreversible actions on the organization or repository settings without the maintainers' consent.
+
+Your responsibility:
+
+- PRs should be up to the contribution standard.
+  - Your code should be properly formatted and linted.
```
Comment (imkiva) [inline .github/CONTRIBUTING.md:44]: Suggestion: Include tests as much as possible if you are implementing new features or fixing bugs.

**Review by imkiva — COMMENTED** [review]
Should we add a line like `XXX should take precedence when YYY` at the end of these files in case there's a situation that has not been taken into account?

Polishing these words is far beyond my level of English. If my text doesn't fit the scene, please trouble me.

**Review by Glavo — APPROVED** [review]

**Comment by ice1000** [comment]
> Should we add a line like XXX should take precedence when YYY at the end of these files in case there's a situation that has not been taken into account?

Do you have specific suggestions? I think right now it's fine and I don't expect a 100%-coverage COC.

**Review by ice1000 — APPROVED** [review]
I addressed mine and @imkiva's comments.

**Comment by ice1000** [comment]
bors delegate=imkiva

**Review by wsx-ucb — APPROVED** [review]
LGTM

**Review by ericwang385 — APPROVED** [review]

**Review by tonfeiz — APPROVED** [review]

**Review by imkiva — APPROVED** [review]
bors r+

**Comment by imkiva** [comment]
> > Should we add a line like XXX should take precedence when YYY at the end of these files in case there's a situation that has not been taken into account?
> 
> Do you have specific suggestions? I think right now it's fine and I don't expect a 100%-coverage COC.

Because I have seen things like "The final interpretation right belongs to XXX" (in Chinese, 最终解释权归 XXX 所有) in many articles of association, I am thinking about whether we also need this _catch-all pattern_ when our community gets bigger. I guess it is only useful legally?

**Comment by ice1000** [comment]
@imkiva I have no idea how to say that and where to say. If you found a way to do it, I'd be happy to see a PR

**Pull request merged**